internal/api/user_devices.go

```diff
@@ -111,6 +112,26 @@ func (s *userDeviceService) ListUserDevicesForUser(ctx context.Context, req *pb.
 	return &pb.ListUserDevicesForUserResponse{UserDevices: out}, nil
 }
 
+func (s *userDeviceService) ListDevicesForWalletAddress(ctx context.Context, req *pb.ListDevicesForWalletAddressRequest) (*pb.ListDevicesForWalletAddressResponse, error) {
+	vNft, err := models.VehicleNFTS(
+		models.VehicleNFTWhere.OwnerAddress.EQ(null.BytesFrom(common.FromHex(req.UserAddress))),
+		qm.Load(models.VehicleNFTRels.UserDevice),
+	).All(ctx, s.dbs().Reader)
+
+	if err != nil {
+		s.logger.Err(err).Str("eth address", req.UserAddress).Msg("Database failure retrieving user's devices.")
+		return nil, status.Error(codes.Internal, "Internal error.")
+	}
+
+	out := make([]*pb.UserDevice, len(vNft))
+
+	for i := 0; i < len(vNft); i++ {
+		out[i] = s.deviceModelToAPI(vNft[i].R.UserDevice)
```

Comment:
Will these guys have `vNft[i].R.UserDevice.R.VehicleNFT` populated? I honestly don't know.

Reply:
It looks like the answer is yes!